components/vni.js

```diff
@@ -0,0 +1,63 @@
+// Vni.js
+
+var _ = require('underscore');
+var Lm = require('./Lm');
+var state = require('./state');
+
+// Hash of vnis
+vnis = []; 
+
+// Define Vni class and its subclasses - InterNodeVni & IipVni
+function Vni( vnid ) {
+  this.vnid = vnid;
+  this.lm = Lm();
+  this.allInputStates = state.allInputStates;
+  this.allOutputStates = state.allOutputStates;
+  this.inputState = state.inputState;
+  this.outputState = state.outputState;
+  this.node = this.nodeInstance;
+}
+
+function NodePortVni( vnid ) {
+  Vni.call(this, vnid);
+  this.previousLms = [];
+}
+
+var IIP_VNID = '*';
+function IipVni() {
```

Comment:
I don't think we need to create a VNI for an IIP, but we  do need to create a State object for an IIP.  If the IIP data is someData, then the State would be:
<pre>
{ lm: Lm(), data: someData }
</pre>